pandas/io/common.py

```diff
@@ -59,10 +59,24 @@ class DtypeWarning(Warning):
     pass
 
 
+class BaseIterator(object):
+    """Subclass this and provide a "__next__()" method to obtain an iterator.
+    Useful only when the object being iterated is non-reusable (e.g. OK for a
+    parser, not for an in-memory table, yes for its iterator)."""
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        raise AbstractMethodError
+
+if not compat.PY3:
+    BaseIterator.next = lambda self : self.__next__()
```

Comment:
you need the lambda for dispatching (otherwise child classes will call the abstract method)